sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -1482,6 +1491,11 @@ private void SetNameFieldsIfNull()
                     // Have to use cancellationToken in body.
                     Debug.Assert(cancellationToken != null || cancellationToken == null);
 
+                    // Have to use async in body.
+#pragma warning disable AZC0109 // Misuse of 'async' parameter.
+                    Debug.Assert(async || !async);
```

Comment:
It's easier to just suppress the unused param warning instead.

Reply:
Fixed in next iteration.